pkg/fleetautoscalers/controller.go

```diff
@@ -293,3 +336,68 @@ func (c *Controller) updateStatusUnableToScale(ctx context.Context, fas *autosca
 
 	return nil
 }
+
+// createFasThread creates a FleetAutoScaler sync routine
+func (c *Controller) createFasThread(ctx context.Context, fas *autoscalingv1.FleetAutoscaler) error {
+	key, err := cache.MetaNamespaceKeyFunc(fas)
+	if err != nil {
+		return err
+	}
+	ticker := c.clock.NewTicker(time.Duration(fas.Spec.Sync.FixedInterval.Seconds) * time.Second)
+	c.fasThreads[key] = fasThread{
+		terminateSignal: make(chan struct{}),
+		generation:      fas.Generation,
+	}
+	// scale immediately when an FAS is created or updated
+	if err := c.fleetAutoScale(ctx, key); err != nil {
+		return err
+	}
+
+	go func() {
+		for {
+			select {
+			case <-c.fasThreads[key].terminateSignal:
+				ticker.Stop()
+				return
+			case <-ticker.C():
+				c.loggerForFleetAutoscalerKey(key).Debug("run routine sync")
+				err := c.fleetAutoScale(ctx, key)
+				if err != nil {
+					c.loggerForFleetAutoscaler(fas).WithError(err).Error("routine sync failed")
+				}
+			}
+		}
+	}()
+	return nil
+}
+
+// removeFasThread remove a FleetAutoScaler sync routine
+func (c *Controller) removeFasThread(fas *autoscalingv1.FleetAutoscaler) {
+	key, err := cache.MetaNamespaceKeyFunc(fas)
+	if err != nil {
+		return
+	}
+	if _, ok := c.fasThreads[key]; ok {
+		c.fasThreads[key].terminateSignal <- struct{}{}
+		delete(c.fasThreads, key)
+	}
+}
+
+// syncFleetAutoscalerWithCustomSyncInterval syncs the fleet based on a custom interval configured in Fas
+func (c *Controller) syncFleetAutoscalerWithCustomSyncInterval(ctx context.Context, key string) error {
+	fas, err := c.getFleetAutoscalerByKey(key)
+	if fas == nil {
```

Comment:
This should check (either instead of or in addition to this) whether err != nil.

Reply:
Thanks for point out. Updated.